rmw_dds_common/include/rmw_dds_common/qos.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RMW_DDS_COMMON__QOS_HPP_
+#define RMW_DDS_COMMON__QOS_HPP_
+
+#include "rmw/qos_profiles.h"
+#include "rmw/types.h"
+
+#include "rmw_dds_common/visibility_control.h"
+
+namespace rmw_dds_common
+{
+
+/// Check if two QoS profiles are compatible
+/**
+ * Two QoS profiles are compatible if a publisher and subcription
+ * using the QoS policies can communicate with each other.
+ *
+ * This implements the rmw API \ref rmw_qos_profile_check_compatible().
+ * See \ref rmw_qos_profile_check_compatible() for more information.
+ *
+ * \param[in] publisher_profile: The QoS profile used for a publisher.
+ * \param[in] subscription_profile: The QoS profile used for a subscription.
+ * \param[out] compatibility: `RMW_QOS_COMPATIBILITY_OK` if the QoS profiles are compatible, or
+ *   `RMW_QOS_COMPATIBILITY_WARNING` if the QoS profiles might be compatible, or
+ *   `RMW_QOS_COMPATIBILITY_ERROR` if the QoS profiles are not compatible.
+ * \param[out] reason: A detailed reason for a QoS incompatibility.
```

Comment:
I noticed that the test code is initializing the 'reason' buffer before calling this function:

`    char reason[2048] = "";`


If this is a requirement for client code, it could be mentioned here. Otherwise, the implementation could also initialize the buffer at the start in case the user declares the buffer on the stack without initialization:

`    char reason[2048];`

and the function appends the message to some junk on the stack.

Reply:
Good catch! I think doing the zero-initialization of the buffer in the function makes sense; one less thing for the user to worry about.

Reply:
See 530c42f